### PR TITLE
[Fix][Validator] Resolve op classes in multi-class files by PascalCase convention

### LINKED ISSUE #803: [BUG][Validator] Fix _resolve_op_class fallback when source file has multiple Op classes
# Description

## Symptom / Motivation
Discovered during PR #798 ([Feat][Reduction] Migrate remaining 17 reduction ops to spec-conformant interface).

`_resolve_op_class("tileops/ops/reduction/reduce.py", "var_fwd")` resolves to `AmaxOp` instead of `VarOp`. The name-matching heuristic splits `var_fwd` into `["var", "fwd"]` and checks `all(part in cls_lower for part in parts)`. No class name contains `"fwd"`, so all candidates fail. The `_fwd` suffix fallback also fails (no class has `fwd` in its name). Final fallback returns `candidates[0]` — alphabetically `AmaxOp`.

This caused a false CI failure: validator checked `AmaxOp.__init__` for `correction` param (which only exists on `_WelfordReduceOp`).

## Root Cause Analysis
The resolver was designed for the common case where each source file has one Op class, or classes include `Fwd`/`Bwd` suffixes. Reduction ops break both assumptions: one file has 10+ Op classes, none have `Fwd`/`Bwd` in their names.

Currently masked because reduction ops are `spec-only` (L1 signature check skipped), but will block any future status promotion.

## Related Files
- `scripts/validate_manifest.py` — `_resolve_op_class()` at line ~378
- `tileops/ops/reduction/reduce.py` — multi-class source file

# Goal
Make `_resolve_op_class` correctly resolve op names to classes when source files contain multiple Op classes without `Fwd`/`Bwd` naming convention.

# Plan

1. After existing heuristics fail, try stripping `_fwd`/`_bwd` suffix from op_name and converting remainder to PascalCase + "Op" (e.g., `var_fwd` → `VarOp`, `var_mean_fwd` → `VarMeanOp`)
2. Add unit tests for multi-class file resolution in `tests/test_validate_manifest.py`
3. Verify all 17 reduction ops resolve correctly

# Constraints
- Must not regress changes from PR #798
- Must not weaken existing validation rules (per manifest-validator domain rules)
- This is validator infrastructure — PR should be scoped to validator files only

# Acceptance Criteria
- [ ] AC-1: `_resolve_op_class` returns correct class for all 17 reduction ops
- [ ] AC-2: Existing resolver behavior unchanged for single-class files and Fwd/Bwd-suffixed classes
- [ ] AC-3: Unit tests cover multi-class resolution scenarios

### PR DESCRIPTION
**Pull request by lcy-seso**
## Summary

Fix `_resolve_op_class` to correctly resolve op names to classes when source files contain multiple Op classes (e.g., `reduce.py` with 8 ops). The previous substring heuristic (`all parts in cls_lower`) was too permissive — `var_fwd` matched `AmaxOp` instead of `VarOp` because no class contained `"fwd"`, and the alphabetical fallback picked the wrong candidate.

Closes #803

### Resolver heuristic rewrite (`scripts/validate_manifest.py`)
- **Removed** the loose substring-based multi-candidate heuristic (`all(p in cls_lower for p in parts)`)
- **Added** exact PascalCase match as the primary multi-candidate strategy: strip `_fwd`/`_bwd` suffix, convert remainder to PascalCase + `"Op"` (e.g., `var_fwd` → `VarOp`, `var_mean_fwd` → `VarMeanOp`)
- **Added** full-name PascalCase fallback for Fwd/Bwd-suffixed class naming conventions (e.g., `batchnorm_fwd` → `BatchNormFwdOp`)
- Retained suffix matching (`fwd`/`bwd` in class name) as final fallback

### Test coverage (`tests/test_validate_manifest.py`)
- Added `TestResolveOpClass` class with 21 test cases:
  - 8 parametrized tests for `reduce.py` multi-class resolution (SumOp, MeanOp, AmaxOp, AminOp, ProdOp, VarOp, StdOp, VarMeanOp)
  - 11 parametrized tests for all single-file reduction ops (argmax, argmin, all, any, count_nonzero, l1_norm, l2_norm, inf_norm, softmax, log_softmax, logsumexp)
  - Single-class file resolution, import error handling, no-op-class handling

### Review fixes
- Removed duplicate test method flagged by Copilot review
- Fixed misleading comment example (`"batchnorm_fwd" → "BatchnormFwdOp"` → `"sum_fwd" → "SumFwdOp"`)

## Test plan

- [x] **AC-1**: `_resolve_op_class` returns correct class for all 19 reduction ops (8 multi-class in `reduce.py` + 11 single-file) — verified by parametrized tests
- [x] **AC-2**: Existing resolver behavior unchanged for single-class files and Fwd/Bwd-suffixed classes — verified by `test_single_class_file_resolves`
- [x] **AC-3**: Unit tests cover multi-class resolution scenarios — 21 test cases in `TestResolveOpClass`

**Test results:** 72 passed, 0 failed

## Follow-up

- #810 — Emit warning on silent `candidates[0]` fallback
- #811 — Deduplicate PascalCase resolution loops

### CODE REVIEW

File: scripts/validate_manifest.py [hunk 1]
```diff
@@ -410,16 +410,26 @@ def _resolve_op_class(op_file: str, op_name: str) -> _ResolveResult:
...
-        if all(p in cls_lower for p in parts):
+        if cls.__name__ == pascal:
             return _ResolveResult(cls=cls)
 
-    # Fallback: use suffix matching for fwd/bwd
+    # Fallback: full op_name as PascalCase (e.g., "batchnorm_fwd" -> "BatchnormFwdOp")
+    full_pascal = "".join(part.capitalize() for part in op_name.split("_")) + "Op"
+    for cls in candidates:
+        if cls.__name__ == full_pascal:
+            return _ResolveResult(cls=cls)
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply scripts/validate_manifest.py:430]: Reviewer determined this is a false positive:

this is a minor refactor suggestion, not a correctness, coverage, or format defect that needs routing

Current HEAD: 6bee85f

File: tests/test_validate_manifest.py [hunk 2]
```diff
@@ -792,6 +792,95 @@ def test_check_op_rejects_empty_equals_value(self, validator):
...
+        assert result.cls.__name__ == "SoftmaxOp"
+
+    def test_fwd_suffix_resolves_single_class_file(self, validator):
+        """A _fwd op name resolves correctly for a single-class file."""
+        result = validator._resolve_op_class(
+            "tileops/ops/reduction/softmax.py", "softmax_fwd",
+        )
+        assert result.cls is not None
+        assert result.cls.__name__ == "SoftmaxOp"
+
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply tests/test_validate_manifest.py]: Addressed in 6bee85f.

Evidence:
- Removed `test_fwd_suffix_resolves_single_class_file` which was identical to `test_single_class_file_resolves`
- Diff: `tests/test_validate_manifest.py` — 8 lines removed (the duplicate test method)

Current HEAD: 6bee85f

File: scripts/validate_manifest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply scripts/validate_manifest.py]: Addressed in 6bee85f.

Evidence:
- Changed comment example from `"batchnorm_fwd" -> "BatchnormFwdOp"` to `"sum_fwd" -> "SumFwdOp"`
- The new example correctly demonstrates the capitalize-per-segment behavior since `sum` is a single segment

Current HEAD: 6bee85f
